### PR TITLE
[MODDCB-20]-Added Checkout status process.

### PR DESCRIPTION
**Pull request by SinghAdes**
## Purpose
[MODDCB-20](https://issues.folio.org/browse/MODDCB-20)
[MODDCB-12](https://issues.folio.org/browse/MODDCB-12)
## Approach
_How does this change fulfill the purpose?_

#### TODOS and Open Questions
- [ ] Use GitHub checklists. When solved, check the box and explain the answer.

## Learning
_Describe the research stage. Add links to blog posts, patterns, libraries or addons used to solve this problem._

### CODE REVIEW

**Comment by SinghAdes** [comment]
**PLEASE ignore code smell , its because we are using switch with 1 case only . it will be fixed automatically in future when we will have more roles ,**

File: src/main/java/org/folio/dcb/listener/CirculationCheckInEventListener.java [hunk 1]
```diff
@@ -2,29 +2,47 @@
...
+
 import static org.folio.dcb.utils.TransactionHelper.getHeaderValue;
+import static org.folio.dcb.utils.TransactionHelper.parseCheckInEvent;
 
 @Log4j2
 @Component
 @RequiredArgsConstructor
 public class CirculationCheckInEventListener {
-  public final LibraryService libraryService;
+  private final LibraryService libraryService;
```
Comment (Vignesh-kalyanasundaram) [inline src/main/java/org/folio/dcb/listener/CirculationCheckInEventListener.java:22]: Please rename this to lendingLibraryService and inject that bean or else we need to change this later When we implement other library roles

File: src/main/java/org/folio/dcb/service/impl/TransactionsServiceImpl.java [hunk 2]
```diff
@@ -63,13 +63,10 @@ public TransactionStatusResponse getTransactionStatusById(String dcbTransactionI
 
   private TransactionStatusResponse generateTransactionStatusResponseFromTransactionEntity(TransactionEntity transactionEntity) {
     TransactionStatus.StatusEnum transactionStatus = transactionEntity.getStatus();
-    Role.TransactionRoleEnum transactionRole = transactionEntity.getRole();
```
Comment (Vignesh-kalyanasundaram) [inline src/main/java/org/folio/dcb/service/impl/TransactionsServiceImpl.java:66]: In get API, I think we need to return both status and role. So we need to keep this line.

File: src/main/java/org/folio/dcb/service/impl/TransactionsServiceImpl.java [hunk 3]
```diff
@@ -63,13 +63,10 @@ public TransactionStatusResponse getTransactionStatusById(String dcbTransactionI
...
     TransactionStatusResponse.StatusEnum transactionStatusResponseStatusEnum = TransactionStatusResponse.StatusEnum.fromValue(transactionStatus.getValue());
-    TransactionStatusResponse.TransactionRoleEnum transactionStatusResponseRoleEnum = TransactionStatusResponse.TransactionRoleEnum.fromValue(transactionRole.getValue());
 
     return TransactionStatusResponse.builder()
       .status(transactionStatusResponseStatusEnum)
-      .transactionRole(transactionStatusResponseRoleEnum)
```
Comment (Vignesh-kalyanasundaram) [inline src/main/java/org/folio/dcb/service/impl/TransactionsServiceImpl.java:72]: We need this as well

File: src/main/java/org/folio/dcb/service/impl/LendingLibraryServiceImpl.java [hunk 4]
```diff
@@ -71,46 +66,28 @@ private void saveDcbTransaction(String dcbTransactionId, DcbTransaction dcbTrans
...
   @Override
   public void updateTransactionStatus(TransactionEntity dcbTransaction, TransactionStatus transactionStatus) {
-    log.info("updateTransactionStatus:: updating dcbTransaction {} to status {} ", dcbTransaction, transactionStatus);
+    log.debug("updateTransactionStatus:: Updating dcbTransaction {} to status {} ", dcbTransaction, transactionStatus);
     if (TransactionStatus.StatusEnum.OPEN.equals(dcbTransaction.getStatus()) && TransactionStatus.StatusEnum.AWAITING_PICKUP.equals(transactionStatus.getStatus())) {
+      log.info("updateTransactionStatus:: Checking in item by barcode: {} ", dcbTransaction.getPatronBarcode());
       circulationService.checkInByBarcode(dcbTransaction);
-      dcbTransaction.setStatus(transactionStatus.getStatus());
-      transactionRepository.save(dcbTransaction);
+    } else if (TransactionStatus.StatusEnum.AWAITING_PICKUP.equals(dcbTransaction.getStatus()) && TransactionStatus.StatusEnum.ITEM_CHECKED_OUT.equals(transactionStatus.getStatus())) {
```
[the remark this answers is not in the document]
Comment (SinghAdes) [reply src/main/java/org/folio/dcb/service/impl/LendingLibraryServiceImpl.java]: Yes , you are right == is more efficient than .equals here as we are comparing Enums. thanks
Fixed

File: src/main/java/org/folio/dcb/listener/CirculationCheckInEventListener.java (hunk 1 above)
Comment (SinghAdes) [reply src/main/java/org/folio/dcb/listener/CirculationCheckInEventListener.java]: Fixed

File: src/main/java/org/folio/dcb/service/impl/TransactionsServiceImpl.java (hunk 2 above)
Comment (SinghAdes) [reply src/main/java/org/folio/dcb/service/impl/TransactionsServiceImpl.java:66]: fixed

File: src/main/java/org/folio/dcb/service/impl/TransactionsServiceImpl.java (hunk 3 above)
Comment (SinghAdes) [reply src/main/java/org/folio/dcb/service/impl/TransactionsServiceImpl.java:72]: fixed

**Review by Vignesh-kalyanasundaram — APPROVED** [review]

**Review by gurleenkaurbp — APPROVED** [review]

**Review by DmytroBykov1 — APPROVED** [review]

**Pull request merged**